### DISCUSSION TITLE
Is it possible to define/constrain the JSON response more specifically

### DISCUSSION THREAD

**Discussion by jazelly** [withcatai/node-llama-cpp#209] [General]
The support of GGML BNF is awesome. Is there any chance to specify the grammer even at more specific level, e.g.
```
interface Field {
  name: string;
  type: 'single-choice' | 'multiple-choice' | 'input';
}
// instead of 
interface Field {
  name: string;
  type: string;
}
```
If not at the GGML level, had anyone tried the similar thing at application level, perhaps with prompting to limit the answer to a few options?

**Comment by giladgd** [comment]
This is possible by using a [JSON schema grammar](https://withcatai.github.io/node-llama-cpp/guide/grammar#using-a-json-schema-grammar)
